src/ffi/revocation.rs

```diff
@@ -75,14 +75,14 @@ pub extern "C" fn anoncreds_update_revocation_status_list(
             Some(timestamp as u64)
         };
         let revoked: Option<BTreeSet<u32>> = if revoked.is_empty() {
-            Some(revoked.as_slice().iter().map(|r| *r as u32).collect())
-        } else {
             None
+        } else {
+            Some(revoked.as_slice().iter().map(|r| *r as u32).collect())
         };
         let issued: Option<BTreeSet<u32>> = if issued.is_empty() {
-            Some(issued.as_slice().iter().map(|r| *r as u32).collect())
-        } else {
             None
+        } else {
+            Some(issued.as_slice().iter().map(|r| *r as u32).collect())
```

Comment:
I believe this condition was flipped around; if the revoked list was empty, return the list mapped to a BTreeSet<u32>, else, return None. So we were always getting back empty revoked lists.

Reply:
@dbluhm @blu3beri you are correct!